nucypher/blockchain/eth/providers.py

```diff
@@ -140,3 +151,77 @@ def _get_test_geth_parity_provider(provider_uri):
 def _get_tester_ganache(provider_uri=None):
     endpoint_uri = provider_uri or 'http://localhost:7545'
     return HTTPProvider(endpoint_uri=endpoint_uri)
+
+
+def make_rpc_request_with_retry(provider: BaseProvider,
+                                should_retry: Callable[[RPCResponse], bool],
+                                logger: Logger = None,
+                                num_retries: int = 3,
+                                exponential_backoff: bool = True,
+                                *args,
+                                **kwargs) -> RPCResponse:
+    response = provider.make_request(*args, **kwargs)
+    if should_retry(response):
```

Comment:
A small nit: I don't care for the word "should" - I'm not convinced it's a well-defined part of speech in English.

Why not `will_retry`?  Or simply, `retry`, as if in the imperative mood?

Reply:
✔️